include/etl/iterator.h

```diff
@@ -604,7 +604,191 @@ namespace etl
     return etl::move_iterator<TIterator>(itr);
   }
 
-#endif
+#endif //ETL_USING_CPP11
+
+  //***************************************************************************
+  // back_insert_iterator
+  //***************************************************************************
+
+  /**
+   * @brief  Turns assignment into insertion.
+   *
+   * These are output iterators, constructed from a container-of-T.
+   * Assigning a T to the iterator appends it to the container using push_back.
+   *
+   * @tparam TContainer
+   */
+  template <class TContainer>
+  class back_insert_iterator : public iterator<output_iterator_tag, void, void, void, void>
+  {
+  public:
+    /// A nested typedef for the type of whatever container you used.
+    typedef TContainer container_type;
+
+    /// The only way to create this %iterator is with a container.
+    explicit ETL_CONSTEXPR17 back_insert_iterator(TContainer& c)
+      : container(addressof(c))
```

Comment:
is it best to use a qualified name here for `addressof()` ?

Reply:
Yes, use `etl::addressof()`.
I prefer to use qualified names, as it protects against weird errors if a header is missing and a global exists with the same name.

Reply:
implemented in b01a9943c88296e579040dd8975003fb95754b30